lib/state.coffee

```diff
@@ -9,15 +10,19 @@ exports.findCommandBySignature = (signature) ->
 	return _.findWhere exports.commands, (command) ->
 		return command.signature.toString() is signature
 
-exports.getMatchCommand = (signature) ->
+exports.getMatchCommand = (signature, callback) ->
 
-	for command in exports.commands
+	# Omit wildcard command from the check
+	commands = _.reject exports.commands, (command) ->
+		return command.isWildcard()
 
-		# Omit wildcard command from the check
-		continue if command.isWildcard()
+	async.eachSeries commands, (command, done) ->
+		command.signature.matches signature, (result) ->
+			return done() if not result
+			return callback(null, command)
```

Comment:
The intention is to break from the loop if there is a result. Maybe there is a better way to express this?